backend/src/main/java/org/eclipse/tractusx/puris/backend/controller/MaterialController.java

```diff
@@ -0,0 +1,139 @@
+        }
+        if (materialService.findByOwnMaterialNumber(materialDto.getOwnMaterialNumber()) != null) {
+            // Cannot create material, ownMaterialNumber is already assigned
+            return new ResponseEntity<>(HttpStatusCode.valueOf(409));
+        }
+        Material createdMaterial;
+        try {
+            createdMaterial = modelMapper.map(materialDto, Material.class);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+
+        createdMaterial = materialService.create(createdMaterial);
+        if (createdMaterial == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(500));
+        }
+
+        return new ResponseEntity<>(HttpStatusCode.valueOf(200));
+    }
+
+    @PutMapping
+    @CrossOrigin
+    @Operation(description = "Updates an existing Material entity with the data given in the request body.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Update was accepted."),
+        @ApiResponse(responseCode = "400", description = "Malformed request body."),
+        @ApiResponse(responseCode = "404", description = "No existing Material Entity found, no update was performed."),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error.")
+    })
+    public ResponseEntity<?> updateMaterial(@RequestBody MaterialEntityDto materialDto) {
+        if (materialDto.getOwnMaterialNumber() == null || materialDto.getOwnMaterialNumber().isEmpty()) {
+            // Cannot update material without ownMaterialNumber
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        Material existingMaterial = materialService.findByOwnMaterialNumber(materialDto.getOwnMaterialNumber());
+        if (existingMaterial == null) {
+            // Cannot update non-existent Material
+            return new ResponseEntity<>(HttpStatusCode.valueOf(404));
+        }
+        Material updatedMaterial;
+        try {
+            updatedMaterial = modelMapper.map(materialDto, Material.class);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        updatedMaterial = materialService.update(updatedMaterial);
+        if (updatedMaterial == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(500));
+        }
+
+        return new ResponseEntity<>(HttpStatusCode.valueOf(200));
+    }
+
+    @GetMapping
+    @CrossOrigin
+    @Operation(description = "Returns the requested Material dto, specified by the given ownMaterialNumber.")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Returns the requested Material."),
+        @ApiResponse(responseCode = "404", description = "Requested Material was not found.")
+    })
+    public ResponseEntity<MaterialEntityDto> getMaterial(@RequestParam String ownMaterialNumber) {
```

Comment:
please add an explanation for the parameter with an example value. Also explain the ownMaterialNumber a little bit. See parameter definition here: https://www.baeldung.com/swagger-parameter-vs-schema

Reply:
Done!